modules/tinymce/src/themes/silver/main/ts/ui/core/color/ColorSwatch.ts

```diff
@@ -86,7 +86,7 @@ const applyColor = (editor: Editor, format: ColorFormat, value: string, onChoice
     const dialog = colorPickerDialog(editor);
     dialog((colorOpt) => {
       colorOpt.each((color) => {
-        Options.addColor(color);
+        Options.addColor(color, format);
```

Comment:
This is flipped around. I think it should be `Options.addColor(format, color);`
I found this by adding a proper type to id making it `'forecolor' | 'hilitecolor' | 'default'` by changing all the IDs to that type.

So I think we should really consider making this a proper type than just a string. That would help out with branching as well forcing one to make sure that all variants are covered if you instead of doing if/else doing if on each of the cases.

Reply:
It was intentionally set as string to allow us or users to reference these fully custom, since that'd help us add more, separate cache's in other places, something we've had a problem with before. Adding the type would effectively make that impossible.

Reply:
Good point the cache could be on custom things I thought we only had bespoke color pickers but if they are to be used on table dialogs I guess then it could be anything. Either way the types found a bug so more tests are needed then to cover everything. In theory the id's could be a wrapped type but seems overkill of the feature is properly covered by tests.

Reply:
I'm surprised none of the existing tests caught this. Don't we have tests to check this?

Reply:
Apparently not but the parameter order here is new code so should be added to this test.